app/tasks.py

```diff
@@ -1,85 +1,44 @@
-from app.base.utils import update_property_images
+from app import celery, s3, redis_client
 from config import S3_BUCKET_CONFIG
 
 bucket = S3_BUCKET_CONFIG["S3_BUCKET"]
 s3_prop_image_dir = S3_BUCKET_CONFIG["PROP_ASSETS"]
-s3_temp_dir = S3_BUCKET_CONFIG["TEMP_DIR"]
 s3_user_image_dir = S3_BUCKET_CONFIG["USER_ASSETS"]
 
 
 @celery.task()
-def process_images(filename, tmp_dir, dir_to_save):
+def image_process(redis_img, upload_dir, filename):
     """
-    This function processes the image files, using the PIL image library. The filename argument is
-    used to create writable binary object, the object is overwritten by the data of the image downloaded
-    from s3 using boto3 client. Then the image processed using PIL is uploaded back to s3 using boto3 client.
-    The name of this object the filename argument. The image is deleted in the temp dir for cleanup.
+    Process image files, using the PIL image library, upload to S3 and delete the image from redis.
     """
-    try:
-        img_obj = open(filename, "wb")
-        s3.download_fileobj(bucket, f"{s3_temp_dir + tmp_dir}{filename}", img_obj)
-        pil_image_obj = Image.open(filename)
-        pil_image_obj.thumbnail((800, 800))
-        pil_image_obj.save(filename)
-
-        with open(filename, "rb") as data_obj:
-            image_uploader(data_obj, s3_prop_image_dir + dir_to_save + filename)
-        os.remove(filename)
-        delete_img_objs(bucket, s3_temp_dir + tmp_dir, file_obj=filename)
-    except Exception as e:
-        return e
-
-
-@celery.task()
-def update_prop_images(temp_dir, image_file_list, prop_id):
-    from app.base.models import Property
-
-    images_list = update_property_images(temp_dir, image_file_list, prop_id)
-    img_list_to_json = json.dumps(images_list)
-    Property.update_property_images(images_list[0], img_list_to_json, prop_id)
-    return "Images Updated"
-
-
-@celery.task()
-def update_profile_image(dir_name, filename, content_type):
-    s3.upload_fileobj(
-        open(filename, "rb"),
-        bucket,
-        f"{s3_user_image_dir}{dir_name}{filename}",
-        ExtraArgs={
-            "ACL": "public-read",
-            "ContentType": content_type
-        }
-    )
-    os.remove(filename)
-
-
-@celery.task()
-def image_uploader(image_file, path_to_save):
+    bytes_img = redis_client.get(redis_img)
+    img = Image.open(io.BytesIO(bytes_img))
+    img.thumbnail((800, 800))
+    img.save(redis_img + ".jpg", format="JPEG")
     try:
         s3.upload_fileobj(
-            image_file,
+            open(redis_img + ".jpg", "rb"),  # This is the name of the file saved on line 23
             bucket,
-            path_to_save,
+            upload_dir + filename,
             ExtraArgs={
                 "ACL": "public-read",
                 "ContentType": "image/jpeg"
             }
         )
-    except Exception as e:
-        return e
+        os.remove(redis_img + ".jpg")  # Delete the file that was saved temporary on local
```

Comment:
same here am expecting us to work with  all image types
perhaps the  extension to come from the configurable and not  hardcoded

Reply:
Noted